### DISCUSSION TITLE
Sometimes, the Luasnip plugin doesn't work properly.

### DISCUSSION THREAD

**Discussion by shinonomeow** [L3MON4D3/LuaSnip#1009] [Q&A] [answered]
I'm not quite sure if this issue is caused by the Luasnip plugin or something else. 
How can I troubleshoot this problem?

Currently, I'm using a Mac system with Neovim 0.9.1,config is lazyvim 
Most of the time, Luasnip works fine, but sometimes it appears and disappears like this.

this is the bug video link
https://www.youtube.com/watch?v=TP9ed3mdd4s

**Comment by shinonomeow** [comment]
I found that I can consistently reproduce it. When I use ":e" or gg to top and then ctrl-o back,
and then insert after   
Link *L = creat_link_tail();
"for() , if or other snippet"

it triggers this bug,it will sellect until the line begin and send a num "1".
```python
#include <stdio.h>
#include <stdlib.h>

struct Link {
  int value;
  Link *next;
};

void del_x(Link *&L, int x) {
  if (!L)
    return;
  Link *pre;
  if (x == L->value) {
    pre = L;
    L = L->next;
    free(pre);
    del_x(L, x);
  } else {
    del_x(L->next, x);
  }
}

Link *creat_link_tail() {
  Link *head = (Link *)malloc(sizeof(Link));
  Link *rear = (Link *)malloc(sizeof(Link));
  head = rear;
  int value;
  scanf("%d", &value);
  while (value != 9999) {
    Link *p = (Link *)malloc(sizeof(Link));
    p->value = value;
    rear->next = p;
    rear = p;
    scanf("%d", &value);
  }
  rear->next = NULL;
  return head;
}

void print_link(Link *L) {
  Link *p = L->next;
  while (p != NULL) {
    printf(" %d ", p->value);
    p = p->next;
    if (p != NULL) {
      printf("->");
    }
  }
  printf("\n");
}

int main(int argc, char *argv[]) {
  freopen("./link_in.txt", "r", stdin);
  Link *a, *b;
  Link *L = creat_link_tail();
  for (range-declaration : range-expression) {
  
  }
  print_link(L);
}
```

**Reply by L3MON4D3** [reply]
Mhmm, haven't seen that one before.. does it also occur if you trigger the snippet manually, so not through `cmp`?

**Reply by shinonomeow** [reply, marked as answer]
I found that it was caused by mini-animate by enabling and disabling all the plugins. Thank you for your help.

**Reply by L3MON4D3** [reply]
Thanks for the follow-up, I'll keep that incompatibility in mind!
